src/theory/arith/nonlinear_extension.cpp

```diff
@@ -793,15 +793,40 @@ bool NonlinearExtension::checkModel(const std::vector<Node>& assertions,
       bool success = true;
       // tf is Figure 3 : tf( x )
       Node atf = d_model.computeConcreteModelValue(tf);
+      Trace("nl-ext-cm-debug")
+          << "Value for is " << tf << " is " << atf << std::endl;
+      Node bl;
+      Node bu;
       if (k == PI)
       {
-        success = d_model.addCheckModelBound(atf, d_pi_bound[0], d_pi_bound[1]);
+        bl = d_pi_bound[0];
+        bu = d_pi_bound[1];
       }
       else if (d_model.isRefineableTfFun(tf))
       {
         d_model.setUsedApproximate();
         std::pair<Node, Node> bounds = getTfModelBounds(tf, d_taylor_degree);
-        success = d_model.addCheckModelBound(atf, bounds.first, bounds.second);
+        bl = bounds.first;
+        bu = bounds.second;
+      }
+      if (!bl.isNull() && !bu.isNull())
+      {
+        // We have rewritten an application of a transcendental function
+        // based on the current model values.It could be that the model value
+        // rewrites sin(x) ---> sin(-c) ---> -sin(c), thus we need
+        // to negate the bounds in this case.
+        if (atf.getKind() != tf.getKind())
+        {
+          if (atf.getKind() == MULT && atf.getNumChildren() == 2
```

Comment:
shouldn't we do this kind of reasoning inside the function `getTfModelBounds`
(https://github.com/CVC4/CVC4/blob/3e91545ab5914fe65f57544d0d00a7dd9e897e64/src/theory/arith/nonlinear_extension.cpp#L3770)

Reply:
I had to refresh my memory on this.  It turns out we don't have to.

In that function, we compute the model value of `f(x)`, viewing it as a term, whereas in the code I changed above, we are computing the value of `f(x^M)` where x^M is the model value of x, which can rewrite to `-sin(c)` for instance.  This isn't necessary here.